packages/orm/src/EntityManager.ts

```diff
@@ -179,7 +181,7 @@ export class EntityManager<C = unknown> {
    */
   #pendingCascadeDeletes: Entity[] = [];
   #dataloaders: Record<string, LoaderCache> = {};
-  #joinRows: Record<string, JoinRow[]> = {};
+  #joinRows: Record<string, JoinRows> = {};
```

Comment:
Instead of having everyone just read/write to a `JoinRow[]` with abandon, starting to abstract a `JoinRows` abstraction around it.